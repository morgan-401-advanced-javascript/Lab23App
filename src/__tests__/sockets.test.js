import React from 'react';
import Sockets from '../components/sockets';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Socket from '../components/sockets';

configure({adapter:new Adapter()});

describe('', () => {
    it('', () => {
        expect(true).toBeTruthy();
    });
});

describe('Socket component', () => {
    it('renders correctly', () => {
        const page = renderer.create(< Socket/>).toJSON();
        expect(page).toMatchSnapshot();
    });
});

describe('Socket Enzyme Testing', () => {
    it('has a Heading', () => {
        let component = mount(<Socket />);
        expect(component.find('div h2').text()).toBe(
            'How is the Weather?',
        );
    })});
describe('Socket Enzyme Testing', () => {
    it('Correct number of Children', () => {
        let component = mount(<Socket />);
        expect(component.find('div').children().length).toEqual(
            4
        );
    })});
