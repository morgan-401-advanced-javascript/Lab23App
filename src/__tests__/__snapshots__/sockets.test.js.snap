// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Socket component renders correctly 1`] = `
<div>
  <h2>
    How is the Weather?
  </h2>
  <p>
    Are we connected to NOAA: 
    false
  </p>
  <p>
    Raining: 
    false
  </p>
  <p>
    Temp: 
    0
  </p>
</div>
`;
